tools/slicec-cs/src/generators/enum_generator.rs

```diff
@@ -72,6 +141,205 @@ fn enum_values(enum_def: &Enum) -> CodeBlock {
     code
 }
 
+fn enumerators_as_nested_records(enum_def: &Enum) -> CodeBlock {
+    let mut code = CodeBlock::default();
+    let namespace = enum_def.namespace();
+
+    for enumerator in enum_def.enumerators() {
+        let escaped_identifier = enumerator.escape_identifier();
+        let mut builder = ContainerBuilder::new(
+            "public partial record class", // Dunet nested records are always public
+            &escaped_identifier,
+        );
+
+        if let Some(summary) = enumerator.formatted_doc_comment_summary() {
+            builder.add_comment("summary", summary);
+        }
+        builder
+            .add_comments(enumerator.formatted_doc_comment_seealso())
+            .add_obsolete_attribute(enumerator)
+            .add_base(enum_def.escape_identifier())
+            .add_fields(enumerator.associated_fields().unwrap_or_default().as_slice());
+
+        // Add cs::attribute
+        for attribute in enumerator.cs_attributes() {
+            builder.add_attribute(attribute);
+        }
+
+        builder.add_block({
+            let mut code = CodeBlock::default();
+            code.writeln(&CommentTag::new(
+                "summary",
+                "The discriminant of this enumerator.".into(),
+            ));
+            writeln!(code, "public new const int Discriminant = {};", enumerator.value());
+            code
+        });
+
+        builder.add_block(
+            FunctionBuilder::new(
+                "internal static",
+                &escaped_identifier,
+                "FromSliceDecoder",
+                FunctionType::BlockBody,
+            )
+            .add_comment(
+                "summary",
+                format!(
+                    r#"Decodes a new instance of <see cref="{escaped_identifier}" /> from the given Slice decoder."#
+                ),
+            )
+            .add_parameter(
+                "ref SliceDecoder",
+                "decoder",
+                None,
+                Some("The Slice decoder.".to_owned()),
+            )
+            .set_body({
+                let mut code = CodeBlock::default();
+
+                if enum_def.is_unchecked {
+                    code.writeln("decoder.SkipSize();");
+                }
+
+                code.writeln(&decode_enum_fields(
+                    enumerator.associated_fields().unwrap_or_default().as_slice(),
+                    &escaped_identifier,
+                    &namespace,
+                    FieldType::NonMangled,
+                    Encoding::Slice2,
+                ));
+
+                // TODO: only for non-compact enum
+                code.writeln("decoder.SkipTagged();");
+
+                code.writeln("return result;");
+                code
+            })
+            .build(),
```

Comment:
Shouldn't be a comma here, since this isn't a parameter or field, just the end of a function chain.
```suggestion
            .build()
```